x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_table.tsx

```diff
@@ -9,36 +9,34 @@ import React from 'react';
 
 import { useActions } from 'kea';
 
-import { EuiBasicTable, EuiBasicTableColumn } from '@elastic/eui';
+import { EuiBasicTable, EuiBasicTableColumn, CriteriaWithPagination } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage, FormattedDate, FormattedNumber } from '@kbn/i18n/react';
 
-import { ENGINES_PAGE_SIZE } from '../../../../../common/constants';
 import { EuiLinkTo } from '../../../shared/react_router_helpers';
 import { TelemetryLogic } from '../../../shared/telemetry';
 import { UNIVERSAL_LANGUAGE } from '../../constants';
 import { ENGINE_PATH } from '../../routes';
 import { generateEncodedPath } from '../../utils/encode_path_params';
 import { EngineDetails } from '../engine/types';
 
-interface EnginesTablePagination {
-  totalEngines: number;
-  pageIndex: number;
-  onPaginate(pageIndex: number): void;
-}
 interface EnginesTableProps {
-  data: EngineDetails[];
-  pagination: EnginesTablePagination;
-}
-interface OnChange {
-  page: {
-    index: number;
+  items: EngineDetails[];
+  loading: boolean;
+  pagination: {
+    pageIndex: number;
+    pageSize: number;
+    totalItemCount: number;
+    hidePerPageOptions: boolean;
   };
+  onChange(criteria: CriteriaWithPagination<EngineDetails>): void;
 }
```

Comment:
I changed these props to better match `<EuiBasicTable>`'s actual props - essentially `<EnginesTable>` should contain all the props you'd pass to an EuiBasicTable sans `columns` (since those are currently shared between the engines table & meta engines table).

Also the typing here was kind of a headache, I tried my best to type `pagination` as `pagination: EuiBasicTableColumn['pagination']` without having to manually repeat stuff but EUI wasn't having it / the types I needed weren't being exported 😢 I banged my face against it for ~5 mins or so before moving on